webapp/channels/src/components/suggestion/suggestion_list.tsx

```diff
@@ -15,48 +14,62 @@ import LoadingSpinner from 'components/widgets/loading/loading_spinner';
 
 // When this file is migrated to TypeScript, type definitions for its props already exist in ./suggestion_list.d.ts.
 
-export default class SuggestionList extends React.PureComponent {
-    static propTypes = {
-        ariaLiveRef: PropTypes.object,
-        inputRef: PropTypes.object,
-        open: PropTypes.bool.isRequired,
-        position: PropTypes.oneOf(['top', 'bottom']),
-        renderDividers: PropTypes.arrayOf(PropTypes.string),
-        renderNoResults: PropTypes.bool,
-        onCompleteWord: PropTypes.func.isRequired,
-        preventClose: PropTypes.func,
-        onItemHover: PropTypes.func.isRequired,
-        pretext: PropTypes.string.isRequired,
-        cleared: PropTypes.bool.isRequired,
-        matchedPretext: PropTypes.array.isRequired,
-        items: PropTypes.array.isRequired,
-        terms: PropTypes.array.isRequired,
-        selection: PropTypes.string.isRequired,
-        components: PropTypes.array.isRequired,
-        suggestionBoxAlgn: PropTypes.object,
+interface Props {
+    ariaLiveRef?: React.Ref<HTMLDivElement>;
+    inputRef?: React.Ref<HTMLInputElement>;
+    open: boolean;
+    position?: 'top' | 'bottom';
+    renderDividers?: string[];
+    renderNoResults?: boolean;
+    onCompleteWord: (term: string, matchedPretext: string, e?: React.KeyboardEventHandler<HTMLDivElement>) => boolean;
+    preventClose?: () => void;
+    onItemHover: (term: string) => void;
+    pretext: string;
+    cleared: boolean;
+    matchedPretext: string[];
+    items: any[];
+    terms: string[];
+    selection: string;
+    components: Array<React.FunctionComponent<any>>;
+    wrapperHeight?: number;
+
+    // suggestionBoxAlgn is an optional object that can be passed to align the SuggestionList with the keyboard caret
+    // as the user is typing.
+    suggestionBoxAlgn?: {
+        lineHeight: number;
+        pixelsToMoveX: number;
+        pixelsToMoveY: number;
     };
+}
 
+export default class SuggestionList extends React.PureComponent<Props> {
     static defaultProps = {
         renderDividers: [],
         renderNoResults: false,
     };
-
-    constructor(props) {
+    contentRef: React.RefObject<HTMLDivElement>;
+    wrapperRef: React.RefObject<HTMLDivElement>;
+    itemRefs: Map<any, any>;
```

Comment:
The key of this map is a "term" and the value , so I think the type of this would be `Map<string, HTMLDivElement>`. If `HTMLDivElement` doesn't work, then `Map<string, any>` would also be fine too since I know a lot of the stuff around this component is difficult to type

Reply:
`Map<string, HTMLDivElement>` doesn't work as a `search_suggestion_list.tsx` complains